sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -138,20 +138,23 @@ case class DataWritingCommandExec(cmd: DataWritingCommand, child: SparkPlan)
  * @param extended whether to do extended explain or not
  * @param codegen whether to output generated code from whole-stage codegen or not
  * @param cost whether to show cost information for operators.
+ * @param optQueryExecution if present, this command uses pre-analyzed plan of it,
+ *                          instead of re-analyzing plan.
  */
 case class ExplainCommand(
     logicalPlan: LogicalPlan,
     extended: Boolean = false,
     codegen: Boolean = false,
-    cost: Boolean = false)
+    cost: Boolean = false,
+    optQueryExecution: Option[QueryExecution] = None)
```

Comment:
I don't have a better way to keep using analyzed plan of dataset and showing correct pre-analyzed plan, other than passing in the query execution of the dataset.